cmake/cmake_options_osx.yml

```diff
@@ -0,0 +1,53 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Instead of having this file, can we add a CMakeList_osx.txt instead? I think this adding another layer on top of CMakeList.txt is unnecessary.

Reply:
That's not how CMake works unfortunately.

Reply:
That's interesting: https://preshing.com/20170511/how-to-build-a-cmake-based-project/

I have been successfully configure CMakeList.txt in this project to build without relying on your script. I have found it much simpler. To build mxnet, I only need the following three lines:

```
mkdir -p build
cd build
cmake -DCMAKE_BUILD_TYPE=Release ..
make
```

Reply:
Please propose or suggest concrete changes. I don't think changing cmakelist is a good idea. Haven't seen that in other projects, if you know better please propose something.

Reply:
We should probably follow what pytorch builds: https://github.com/pytorch/pytorch with a proper setup.py

The current way of dev_menu.py seems like a hacky solution.